share/revng/test/tests/scripting.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import argparse
+import re
+from typing import Callable
+
+from revng import ptml
+from revng.scripting import CLIProfile, DaemonProfile
+from revng.scripting.projects import Project
+
+
+# We patch the `import_and_analyze` method so that it doesn't
+# depend on `renvg-c`.
+def tmp_import_and_analyze(self, input_binary_path: str):
+    self.import_binary(input_binary_path)
+    self.analyses_list("revng-initial-auto-analysis")
+    self._get_pipeline_description()
+    self._set_model_mixins()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--resume", help="Path to the resume directory", required=True)
+    parser.add_argument("--binary", help="Path to the Binary", required=True)
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("--cli", help="Run revng cli", action="store_true")
+    group.add_argument("--daemon", help="Run revng daemon", action="store_true")
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    profile: CLIProfile | DaemonProfile
+
+    if args.cli:
+        profile = CLIProfile()
+    elif args.daemon:
+        profile = DaemonProfile()
+    else:
+        raise ValueError("The script expects either --cli or --daemon")
+
+    run_test(profile.get_project, args.binary, args.resume)
+
+
+def run_test(project_getter: Callable[[str], Project], binary: str, resume: str):
+    project = project_getter(resume)
+    project.import_and_analyze = tmp_import_and_analyze.__get__(project)  # type: ignore
```

Comment:
Drop the use of `__get__`, it's a bit too "magical".
Simply remove the `import_and_analyze` line and replace it with the body of `tmp_import_and_analyze`